components/raftstore/src/errors.rs

```diff
@@ -139,38 +140,72 @@ quick_error! {
     }
 }
 
+pub struct Error(pub Box<ErrorInner>);
```

Comment:
How about just changing some into box? Errors like `NotLeader` are very common and should be small enough already.

Reply:
I found it's hard to re-organize the codes if just wrapping some `Error`s into `Box`. Do you have any suggestions to do this?

Reply:
I think implementing this [RFC](https://github.com/tikv/rfcs/pull/38) can solve these problems altogether.